src/common/legacy/src/transformations/convert_opset1_to_legacy/reshape_fully_connected.cpp

```diff
@@ -28,14 +28,15 @@ ngraph::pass::ReshapeFullyConnected::ReshapeFullyConnected() {
 
         auto input_shape = fc->input_value(0).get_shape();
         auto output_shape = fc->get_shape();
+        auto reshape_dim = fc->input_value(1).get_shape();
```

Comment:
it will fail if shape of 2nd input to FullyConnected is dynamic.
you have to update the pattern with `pattern::has_static_shape()`  for 2nd input.
![image](https://user-images.githubusercontent.com/21181928/204792895-056f5a13-7534-495f-af9b-d61cc493a2d4.png)
and it looks like we can delete this restriction for 1st input because we don't actually need the shape of 1st input, we only use only a rank (line 33). Please change line 29 to  `auto input_shape = fc->input_value(0).get_partial_shape(); `.
and add check that rank is static